### PR TITLE
[main] Remove kernel-hyperv and hyperv kernel option

### PR DESCRIPTION
**Pull request by rlmenge**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Remove the `kernel-hyperv` spec and the option to select "hyperv" in image configurations.

###### Change Log  

- Remove the kernel-hyperv option

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- local package build of kernel, rebuilt go tools, local image gen
- Pipeline id: 170255

### CODE REVIEW

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 1]
```diff
@@ -313,25 +313,10 @@ func PackageNamesFromSingleSystemConfig(systemConfig configuration.SystemConfig)
 func SelectKernelPackage(systemConfig configuration.SystemConfig, isLiveInstall bool) (kernelPkg string, err error) {
 	const (
 		defaultOption = "default"
-		hypervOption  = "hyperv"
 	)
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/installutils/installutils.go:316]: Nit: now we can do a single line of `const hypervOption  = "hyperv"`.

File: toolkit/tools/imagegen/configuration/configuration_test.go [hunk 2]
```diff
@@ -531,7 +529,6 @@ var expectedConfiguration Config = Config{
 			},
 			KernelOptions: map[string]string{
 				"default": "kernel",
-				"hyperv":  "kernel-hyperv",
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/configuration/configuration_test.go:534]: I think we could do a similar update inside `imagegen/attendedinstaller/_manualrun/manualrun.go`.

Comment (rlmenge) [reply toolkit/tools/imagegen/configuration/configuration_test.go:534]: manualrun.go does not contain hyperv options. This file is used as a tool to quickly test the iso installer flow and the package lists are not explicitly checked. However, they may be used as inputs to other tools so I will fix while I am here.

File: toolkit/docs/formats/imageconfig.md [hunk 3]
```diff
@@ -196,27 +196,16 @@ KernelOptions is mandatory for all non-`rootfs` image types.
...
-},
-```
-
-A sample KernelOptions specifying a default kernel and a specialized kernel for Hyper-V scenario:
-
-``` json
-"KernelOptions": {
-    "default": "kernel",
-    "hyperv": "kernel-hyperv"
+    "default": "kernel-rt"
```
Comment (christopherco) [inline toolkit/docs/formats/imageconfig.md:208]:
```suggestion
    "default": "kernel"
```

**Review by PawelWMS — APPROVED** [review]

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 4]
```diff
@@ -311,27 +311,10 @@ func PackageNamesFromSingleSystemConfig(systemConfig configuration.SystemConfig)
...
+	const defaultOption = "default"
 
 	optionToUse := defaultOption
 
-	// Only consider Hyper-V for an ISO
-	if isLiveInstall {
-		// Only check if running on Hyper V if there's a kernel option for it
-		_, found := systemConfig.KernelOptions[hypervOption]
-		if found {
-			isHyperV, err := isRunningInHyperV()
```
Comment (christopherco) [inline toolkit/tools/imagegen/installutils/installutils.go:326]: Can we also remove `isRunningInHyperV`? I don't think anything else calls into it.

**Review by thcrain-msft — APPROVED** [review]
🎉

File: toolkit/docs/formats/imageconfig.md (hunk 3 above)
Comment (rlmenge) [reply toolkit/docs/formats/imageconfig.md]: Done :)

File: toolkit/tools/imagegen/installutils/installutils.go (hunk 4 above)
Comment (rlmenge) [reply toolkit/tools/imagegen/installutils/installutils.go:326]: Agreed. Done :)

**Review by christopherco — APPROVED** [review]
:shipit:

**Pull request merged**